src/backend/mapServer.js

```diff
@@ -0,0 +1,69 @@
+const { promisify } = require("util");
+const getPort = require("get-port");
+const createServer = require("@mapeo/map-server").default;
+const log = require("debug")("map-server");
+
+const AsyncService = require("./upgrade-manager/async-service");
+
+// TODO: Does it make sense for this to extend AsyncService?
+class MapServer extends AsyncService {
+  /** @type {number?} */
+  #port;
+  /** @type {import('fastify').FastifyInstance} */
+  #fastify;
+
+  /** @type {boolean} */
+  #fastifyStarted = false;
+
+  /**
+   * @param {object} object
+   * @param {string} dbPath
+   *
+   */
+  constructor({ dbPath }) {
+    super();
+    this.#fastify = createServer({ logger: false }, { dbPath });
+  }
+
+  /**
+   * Start the server on the specified port. Listen on all interfaces.
+   *
+   * @returns {Promise<void>} Resolves with the port number when server is started
+   */
+  async _start() {
+    if (!this.#port) {
+      this.#port = await getPort();
+    }
+
+    log(`${this.#port}: starting`);
+    if (!this.#fastifyStarted) {
+      log("first start, initializing fastify");
+      await this.#fastify.listen(this.#port);
+      this.#fastifyStarted = true;
+    } else {
+      log("second start, listening");
+      const { server } = this.#fastify;
+      await promisify(server.listen.bind(server))(this.#port);
```

Comment:
```suggestion
      await promisify(server.listen).call(server, this.#port);
```
Nit pick, but I wonder if creating a new bound function each time has more of an overhead that a `call()`?

Reply:
honestly i have no idea. i just copied this from the upgrade example and my JS knowledge isn't esoteric enough to know this off the top of my head 😄

Reply:
upon reading mdn seems like this change makes sense and would help a little